java/integTest/src/test/java/glide/modules/JsonTests.java

```diff
@@ -206,6 +207,86 @@ public void arrappend() {
                 () -> Json.arrappend(client, "non_existing_key", ".b", new String[] {"\"six\""}).get());
     }
 
+    @Test
+    @SneakyThrows
+    public void arrindex() {
+        String key1 = UUID.randomUUID().toString();
+        String key2 = UUID.randomUUID().toString();
+        String key3 = UUID.randomUUID().toString();
+
+        String doc1 =
+                "{\"a\": [1, 3, true], \"b\": {\"a\": [3, 4, [\"value\", 3, false], 5], \"c\": {\"a\":"
+                        + " 42}}}";
+
+        String doc2 =
+                "{\"a\": [1, 3, true, \"foo\", \"meow\", \"m\", \"foo\", \"lol\", false], \"b\": {\"a\":"
+                        + " [3, 4, [\"value\", 3, false], 5], \"c\": {\"a\": 42}, \"empty\": []}}";
+
+        String doc3 = "{\"a\": 123123}";
+
+        assertEquals("OK", Json.set(client, key1, "$", doc1).get());
+        assertArrayEquals(
+                new Object[] {2L, -1L, null}, (Object[]) Json.arrindex(client, key1, "$..a", "true").get());
+
+        assertArrayEquals(
+                new Object[] {1L, 0L, null},
+                (Object[]) Json.arrindex(client, gs(key1), gs("$..a"), gs("3")).get());
+
+        assertEquals("OK", Json.set(client, key2, "$", doc2).get());
+
+        assertArrayEquals(
+                new Object[] {6L, -1L, null},
+                (Object[])
+                        Json.arrindex(client, key2, "$..a", "\"foo\"", new JsonArrindexOptions(6L, 8L)).get());
+
+        assertArrayEquals(
+                new Object[] {-1L, -1L, null},
+                (Object[])
+                        Json.arrindex(client, key2, "$..a", "null", new JsonArrindexOptions(6L, 8L)).get());
+        assertArrayEquals(
+                new Object[] {-1L, -1L, null},
+                (Object[])
+                        Json.arrindex(client, gs(key2), gs("$..a"), gs("null"), new JsonArrindexOptions(6L, 8L))
+                                .get());
+
+        assertArrayEquals(
+                new Object[] {6L, -1L, null},
+                (Object[])
+                        Json.arrindex(
+                                        client, gs(key2), gs("$..a"), gs("\"foo\""), new JsonArrindexOptions(6L, 8L))
+                                .get());
+
+        assertArrayEquals(
+                new Object[] {6L, -1L, null},
+                (Object[])
+                        Json.arrindex(client, key2, "$..a", "\"foo\"", new JsonArrindexOptions(6L)).get());
+
+        // value doesn't exist
+        assertArrayEquals(
+                new Object[] {null},
+                (Object[])
+                        Json.arrindex(client, key1, "$..b", "true", new JsonArrindexOptions(1L, 3L)).get());
+
```

Comment:
* we should add a test where start index > end index
* we should add a test where end index is really big (bigger than the length of the string)